src/test/run-pass/generator/static-generators.rs

```diff
@@ -21,8 +22,7 @@ fn main() {
         yield;
         assert_eq!(b as *const _, &a as *const _);
     };
-    unsafe {
-        assert_eq!(generator.resume(), GeneratorState::Yielded(()));
-        assert_eq!(generator.resume(), GeneratorState::Complete(()));
-    }
+    let mut generator = unsafe { Pin::new_unchecked(&mut generator) };
```

Comment:
Why does this use unsafe code?

Reply:
Constructing a `Pin` to a reference to `!Unpin` value is unsafe. There is a safe [macro-based solution](https://docs.rs/pin-utils/0.1.0-alpha.3/pin_utils/macro.pin_mut.html) to do so on the stack, but it's not included in `std`. I could use a `Pin<Box>` here instead safely, but I thought better to just keep it on the stack.

I can add a safety note about why this is safe (same reason as `pin_mut!` is safe, it shadows the owning binding so the value cannot be moved after this line).

Reply:
Added a note on the safety here.